psa-crypto/src/types/key.rs

```diff
@@ -305,6 +307,50 @@ impl Attributes {
         get_attributes_res?;
         Ok(attributes?)
     }
+
+    /// Sufficient size for a buffer to export the key, if supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn export_key_output_size(self) -> Result<usize> {
+        Attributes::export_key_output_size_base(self.key_type, self.bits)
+    }
+
+    /// Sufficient size for a buffer to export the public key, if supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn export_public_key_output_size(self) -> Result<usize> {
+        match self.key_type {
+            Type::RsaKeyPair
+            | Type::RsaPublicKey
+            | Type::EccKeyPair { .. }
+            | Type::EccPublicKey { .. }
+            | Type::DhKeyPair { .. }
+            | Type::DhPublicKey { .. } => {
+                let pub_type = self.key_type.key_type_public_key_of_key_pair()?;
+                Attributes::export_key_output_size_base(pub_type, self.bits)
+            }
+            _ => Err(Error::InvalidArgument),
+        }
+    }
+
+    /// Sufficient size for a buffer to export the given key type, if supported
+    #[cfg(feature = "with-mbed-crypto")]
+    fn export_key_output_size_base(key_type: Type, bits: usize) -> Result<usize> {
+        let size =
+            unsafe { psa_crypto_sys::PSA_EXPORT_KEY_OUTPUT_SIZE(key_type.try_into()?, bits) };
+        if size > 0 {
+            Ok(size)
+        } else {
+            Err(Error::NotSupported)
+        }
```

Comment:
Same as what Hugues said, this works fine, no need to change, just for reference:
```rust
match unsafe { 
    psa_crypto_sys::PSA_EXPORT_KEY_OUTPUT_SIZE(key_type.try_into()?, bits) 
  } {
    0 => Err(Error::NotSupported),
    val => Ok(val),
}
```

Reply:
Ah yes, I was sure there would be a more succinct way of writing that bit, didn't think of doing it like your example!